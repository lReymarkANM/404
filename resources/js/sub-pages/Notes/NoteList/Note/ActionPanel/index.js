import { FaFileWord, FaPen, FaTrashAlt } from 'react-icons/fa';
import useActionPanel from './hook';
import TargetWordCountInput from './TargetWordCountInput';

const ActionPanel = ({ noteId, wordCount, onOpen, onEdit, onDelete }) => {
  const { isTargetWordCountInputCollapse, setIsTargetWordCountInputCollapse } = useActionPanel();

  return (
    <div className="flex justify-between items-center border-t border-dark-gray h-[40px]">
      <div className="flex items-center">
        <div
          className="text-dark-gray flex items-center cursor-pointer relative"
          onClick={() => setIsTargetWordCountInputCollapse(!isTargetWordCountInputCollapse)}
        >
          <FaFileWord />
          <span className="ml-1 text-[13px] underline decoration-1 active:opacity-50">
            {wordCount}
          </span>
          
          {isTargetWordCountInputCollapse && (
            <TargetWordCountInput
              noteId={noteId}
            />
          )}
        </div>
      </div>
      <div className="flex items-center">
        <button
          onClick={onOpen}
          className="bg-pink text-white flex justify-center items-center w-[50px] h-[25px] rounded-md"
        >
          <span className="text-[10px]">Open</span>
        </button>
        <button
          onClick={onEdit}
          className="text-dark-gray flex items-center ml-4"
        >
          <FaPen />
          <span className="ml-1 text-[10px]">Edit</span>
        </button>
        <button
          onClick={onDelete}
          className="text-dark-gray flex items-center ml-4"
        >
          <FaTrashAlt />
          <span className="ml-1 text-[10px]">Delete</span>
        </button>
      </div>
    </div>
  );
}

export default ActionPanel;