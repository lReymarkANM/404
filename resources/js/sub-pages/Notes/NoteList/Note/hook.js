import { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Cookies from 'js-cookie';
import useNotes from '@sub-pages/Notes/hook';
import useEditor from '@sub-pages/Notes/TextEditor/Editor/hook';
import { showNotification } from '@reducers/notificationSlice';
import { setNote } from '@reducers/noteSlice';

const useNote = note => {
  const collectionId = useSelector(state => state.notes.collectionId);
  const dispatch = useDispatch();
  const { getNotes } = useNotes(collectionId || Cookies.get('collectionId'));
  const { getNote } = useEditor();
  const [title, setTitle] = useState('');
  const [wordCount, setWordCount] = useState(0);
  const [targetWordCount, setTargetWordCount] = useState('');
  const [editModeDisable, setEditModeDisable] = useState(true);
  const [buttonText, setButtonText] = useState('Save');

  useEffect(() => {
    setTitle(note.title);
    setWordCount(getContentWordCount(note.content));
    setTargetWordCount(note.targetWordCount);
  }, []);

  const handleTitleChange = e => {
    setTitle(e.target.value);
    setButtonText('Save');
  }

  const save = async () => {
    setButtonText('Saving...');
    const response = await fetch(`/api/note/${note.id}/update`, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${Cookies.get('token')}`,
        'X-CSRF-TOKEN': document.querySelector('meta[name=csrf_token]').content
      },
      body: JSON.stringify({ title, content: note.content })
    });
    if (!response.ok) return;
    getNotes();
    getNote();
    setEditModeDisable(true);
    setButtonText('Saved!');
    dispatch(showNotification({ label: 'Saved!' }));
  }

  const remove = async () => {
    setButtonText('Saving...');
    const response = await fetch(`/api/note/${note.id}/delete`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${Cookies.get('token')}`,
        'X-CSRF-TOKEN': document.querySelector('meta[name=csrf_token]').content
      }
    });
    if (!response.ok) return;
    getNotes();
    dispatch(showNotification({ label: 'Note deleted' }));
  }

  const selectNote = () => {
    dispatch(setNote({ note }));
    Cookies.set('noteId', note.id, { expires: 3 });
    Cookies.set('showNotesList', true, { expires: 3 });
  }

  const getContentWordCount = content => {
    const wordCount = content ? content.split(' ').length : 0;
    return wordCount;
  }

  return { title, wordCount, targetWordCount, editModeDisable, setEditModeDisable, buttonText, handleTitleChange, save, remove, selectNote };
}

export default useNote;