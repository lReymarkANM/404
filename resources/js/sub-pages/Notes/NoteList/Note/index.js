import useNote from './hook';
import Title from '@sub-pages/Collections/Collection/Title';
import ActionPanel from './ActionPanel';

const Note = ({ data, onOpen }) => {
  const {
    title, handleTitleChange,
    wordCount, targetWordCount,
    editModeDisable, setEditModeDisable,
    buttonText,
    save, remove, selectNote
  } = useNote(data);
  
  return (
    <div className="border border-dark-gray rounded-md px-4 mb-2">
      <Title
        title={title}
        onTitleChange={handleTitleChange}
        onSave={save}
        isEditModeDisable={editModeDisable}
        saveButtonLabel={buttonText}
      />
      <ActionPanel
        noteId={data.id}
        wordCount={wordCount}
        onOpen={() => {
          const DESKTOP_WIDTH = 1280;
          if (window.innerWidth < DESKTOP_WIDTH) {
            onOpen();
            selectNote();
            return;
          };
          selectNote();
        }}
        onEdit={() => setEditModeDisable(!editModeDisable)}
        onDelete={remove}
      />
    </div>
  );
}

export default Note;