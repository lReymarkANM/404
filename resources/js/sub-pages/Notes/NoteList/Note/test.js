import 'regenerator-runtime';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Note from './index';

test('Note renders correctly', () => {
  const data = { id: 1 };
  const initialState = {
    notes: {
      collectionId: 1
    },
    note: {
      data: {}
    }
  };
  const mockStore = configureStore();

  const component = renderer.create(
    <Provider store={mockStore(initialState)}>
      <Note
        data={data}
      />
    </Provider>
  );
  const NoteComponent = component.toJSON();
  expect(NoteComponent).toMatchSnapshot();
});