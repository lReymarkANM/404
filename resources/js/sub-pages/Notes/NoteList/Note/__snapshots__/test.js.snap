// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders correctly 1`] = `
<div
  className="border border-dark-gray rounded-md px-4 mb-2"
>
  <div
    className="min-h-[100px] py-2 flex flex-wrap justify-center items-center cursor-pointer"
  >
    <span
      className="text-[20px] text-center"
    >
      
    </span>
  </div>
  <div
    className="flex justify-between items-center border-t border-dark-gray h-[40px]"
  >
    <div
      className="flex items-center"
    >
      <div
        className="text-dark-gray flex items-center cursor-pointer relative"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm57.1 120H305c7.7 0 13.4 7.1 11.7 14.7l-38 168c-1.2 5.5-6.1 9.3-11.7 9.3h-38c-5.5 0-10.3-3.8-11.6-9.1-25.8-103.5-20.8-81.2-25.6-110.5h-.5c-1.1 14.3-2.4 17.4-25.6 110.5-1.3 5.3-6.1 9.1-11.6 9.1H117c-5.6 0-10.5-3.9-11.7-9.4l-37.8-168c-1.7-7.5 4-14.6 11.7-14.6h24.5c5.7 0 10.7 4 11.8 9.7 15.6 78 20.1 109.5 21 122.2 1.6-10.2 7.3-32.7 29.4-122.7 1.3-5.4 6.1-9.1 11.7-9.1h29.1c5.6 0 10.4 3.8 11.7 9.2 24 100.4 28.8 124 29.6 129.4-.2-11.2-2.6-17.8 21.6-129.2 1-5.6 5.9-9.5 11.5-9.5zM384 121.9v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
          />
        </svg>
        <span
          className="ml-1 text-[13px] underline decoration-1 active:opacity-50"
        >
          0
        </span>
      </div>
    </div>
    <div
      className="flex items-center"
    >
      <button
        className="bg-pink text-white flex justify-center items-center w-[50px] h-[25px] rounded-md"
        onClick={[Function]}
      >
        <span
          className="text-[10px]"
        >
          Open
        </span>
      </button>
      <button
        className="text-dark-gray flex items-center ml-4"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          />
        </svg>
        <span
          className="ml-1 text-[10px]"
        >
          Edit
        </span>
      </button>
      <button
        className="text-dark-gray flex items-center ml-4"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          />
        </svg>
        <span
          className="ml-1 text-[10px]"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</div>
`;
